src/context/trace_context.rs

```diff
@@ -119,8 +118,16 @@ impl TracingContext {
     }
 
     #[inline]
-    pub fn trace_id(&self) -> &str {
-        &self.inner.trace_id
+    pub fn trace_id(&self) -> String {
+        self.with_trace_id(ToString::to_string)
+    }
+
+    fn with_trace_id<T>(&self, f: impl FnOnce(&String) -> T) -> T {
+        f(&*self.inner.trace_id.try_read().expect(LOCK_MSG))
+    }
```

Comment:
Are you going to expose trace ID to users for an override?

Reply:
The `trace_id()` method return a copy of `trace_id`, the original will not be override, the `with_trace_id` is private.